src/basic/path-util.c

```diff
@@ -814,3 +816,57 @@ bool is_device_path(const char *path) {
                 path_startswith(path, "/dev/") ||
                 path_startswith(path, "/sys/");
 }
+
+int systemd_installation_has_version(const char *root, unsigned minimal_version) {
+        _cleanup_free_ char *path = NULL;
+        _cleanup_strv_free_ char **names = NULL;
+        char **name;
+        int r;
+
+        /* Try to guess if systemd installation is later than the specified version. This
+         * is hacky and likely to yield false negatives, particularly if the installation
+         * is non-standard. False positives should be relatively rare.
+         */
+
+        path = prefix_root(root, "lib/systemd/libsystemd-shared-*.so");
```

Comment:
I thought we always install systemd in `/usr/lib`, and `libsystemd-shared` also ends up in the non-arch-specific path. This is true for Fedora, Debian, Ubuntu. I don't see a Mageia systemd 231 package, but in 230, they are also using `/usr/lib/systemd/`. *Is* anyone using `/usr/lib64/systemd/`?